pkg/service/form_submission.go

```
@@ -335,3 +337,32 @@ func AddFlagToFormSubmission(context *web.AppContext) (*model.FormSubmission, *w
 	return &s, nil
 
 }
+
+func SearchFormSubmissions(c *web.AppContext) ([]model.FormSubmission, *web.AppError) {
+
+	// ensure that the text index is created
+	index := mgo.Index{
+		Key:        []string{"$text:replies.answer"},
+		Unique:     false,
+		DropDups:   false,
+		Background: false,
+		Sparse:     true,
+		Name:       "replies.answer.text",
+	}
+	_ = c.MDB.DB.C(model.FormSubmissions).EnsureIndex(index)
```

it should return an error